From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 4 Jan 2026 13:07:26 +0100
Subject: mixer-stream: Allow ports to not exist
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

With pipewire 1.5.84 the port list might change underneath us,
see

   https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/5053

We thus can't assume anymore that we find a given port.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Origin: https://gitlab.gnome.org/GNOME/libgnome-volume-control/-/merge_requests/31
Bug: https://gitlab.gnome.org/GNOME/libgnome-volume-control/-/issues/34
Bug-Debian: https://bugs.debian.org/1126514
Applied-upstream: gnome-shell 50, commit:798230599950607ec285daecd070f5bcb28580e4
---
 subprojects/gvc/gvc-mixer-stream.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/subprojects/gvc/gvc-mixer-stream.c b/subprojects/gvc/gvc-mixer-stream.c
index f9bcc40..0382817 100644
--- a/subprojects/gvc/gvc-mixer-stream.c
+++ b/subprojects/gvc/gvc-mixer-stream.c
@@ -532,8 +532,6 @@ gvc_mixer_stream_get_port (GvcMixerStream *stream)
                 }
         }
 
-        g_assert_not_reached ();
-
         return NULL;
 }
 
